tests/unit_tests/dft/include/compute_out_real_real.hpp

```diff
@@ -0,0 +1,96 @@
+#ifndef ONEMKL_COMPUTE_OUT_REAL_REAL_HPP
+#define ONEMKL_COMPUTE_OUT_REAL_REAL_HPP
+
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+#include "oneapi/mkl.hpp"
+#include "test_helper.hpp"
+#include "../../blas/include/test_common.hpp"
+#include "test_common.hpp"
+#include "oneapi/mkl/dft/descriptor.hpp"
+#include "oneapi/mkl/dft/types.hpp"
+#include "reference_dft.hpp"
+
+/* Test is not implemented because currently there are no available implementation that support it */
+template <dft::precision precision, dft::domain domain>
+int DFT_Test<precision, domain>::test_out_of_place_real_real_USM() {
+    if (!init()) {
+        return test_skipped;
+    }
+
+    dft::descriptor<precision, domain> descriptor{ default_1d_lengths };
+    try {
+        descriptor.set_value(oneapi::mkl::dft::config_param::PLACEMENT,
+                             oneapi::mkl::dft::config_value::INPLACE);
+        descriptor.set_value(oneapi::mkl::dft::config_param::COMPLEX_STORAGE,
+                             oneapi::mkl::dft::config_value::REAL_REAL);
+        descriptor.commit(sycl_queue);
+    }
+    catch (oneapi::mkl::unimplemented &e) {
+        std::cout << "Skipping test because: \"" << e.what() << "\"" << std::endl;
+        return test_skipped;
+    }
+
+    return test_skipped;
+}
+
+/* Test is not implemented because currently there are no available implementation that support it */
+template <dft::precision precision, dft::domain domain>
+int DFT_Test<precision, domain>::test_out_of_place_real_real_buffer() {
+    if (!init()) {
+        return test_skipped;
+    }
+
+    dft::descriptor<precision, domain> descriptor{ default_1d_lengths };
+    try {
+        descriptor.set_value(oneapi::mkl::dft::config_param::PLACEMENT,
+                             oneapi::mkl::dft::config_value::INPLACE);
+        descriptor.set_value(oneapi::mkl::dft::config_param::COMPLEX_STORAGE,
+                             oneapi::mkl::dft::config_value::REAL_REAL);
+        descriptor.commit(sycl_queue);
+    }
+    catch (oneapi::mkl::unimplemented &e) {
+        std::cout << "Skipping test because: \"" << e.what() << "\"" << std::endl;
+        return test_skipped;
+    }
+
+    //    sycl::buffer<PrecisionType, 1> in_dev_re{ sycl::range<1>(default_1d_lengths) };
```

Comment:
I wrote this implementation before realizing that it is not supported at the moment. I'm might be using wrong assumption and most likely the test will not work out of the box. Not sure if I should keep it or just delete. Any opinions?

Reply:
You are right. It should be NOT_INPLACE here. A case of bad copy-paste. However, that's not the issue. The issue is that real_real functions are not currently supported. 

So I have no way of validating that the commented code is correct. I'm thinking that maybe I should just check that compute_forward / compute_backwards throws an exception and remove the validation code.

Reply:
Maybe but I'm wondering is it the combinaison of options `NOT_INPLACE` and `REAL_REAL` that is unsupported? Do you get an error message with more details or a backtrace? Is it just one backend that doesn't support this or none of them?

Reply:
Sorry I can see now you answered that real_real is not supported. I don't have a strong opinion then if you want to remove this or always skip the test.